File: php/src/vaas/Message/VaasVerdict.php
<?php

namespace VaasSdk\Message;

class VaasVerdict
{
    public function __construct(VerdictResponse $verdictResponse)
    {
        $this->Sha256 = $verdictResponse->sha256 ?? "";
        $this->Verdict = $verdictResponse->verdict ?? Verdict::UNKNOWN;
        $this->Guid = $verdictResponse->guid ?? "";
        $this->MimeType = $verdictResponse->mime_type ?? null;
        $this->FileType = $verdictResponse->file_type ?? null;
        $this->Detection = $verdictResponse->detection ?? null;
    }

    public string $Sha256;
    public Verdict $Verdict;
    public string $Guid;
    public ?string $FileType;
    public ?string $MimeType;
    public ?string $Detection;
}

File: python/src/vaas/messages/vaas_verdict.py
from typing import Literal, Optional
from pydantic import BaseModel

class VaasVerdict(BaseModel):
    sha256: str
    verdict: Literal["Clean", "Malicious", "Unknown", "Pup"]
    detection: Optional[str]
    fileType: Optional[str]
    mimeType: Optional[str]

    @staticmethod
    def from_report(report):
        return VaasVerdict(
            sha256=report.sha256,
            verdict=report.verdict,
            detection=report.detection,
            fileType=report.fileType,
            mimeType=report.mimeType
        )
